gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -1311,6 +1311,135 @@ function plymeta:SafeDropWeapon(wep, keepSelection)
 	return true
 end
 
+local function TraceAmmoDrop(ply)
+	local pos, ang = ply:GetShootPos(), ply:EyeAngles()
+	local fwd, rgt, up = ang:Forward(), ang:Right(), ang:Up()
+
+	local dir = fwd * 32
+	rgt:Mul(6)
+	up:Mul(-5)
+	dir:Add(rgt)
+	dir:Add(up)
+
+	local tr = util.QuickTrace(pos, dir, ply)
+
+	return tr, pos, dir, fwd
+end
+
+---
+-- Checks if the ammo can be dropped in a safe manner.
+-- @param Weapon wep The weapon's ammo that should be dropped
+-- @return boolean Returns if this weapon's ammo can be dropped
+-- @realm server
+function plymeta:CanSafeDropAmmo(wep)
+	if not IsValid(self) or not (IsValid(wep) and wep.AmmoEnt) then
+		return false
+	end
+
+	local tr = TraceAmmoDrop(self)
+
+	if tr.HitWorld then
+		LANG.Msg(self, "drop_no_room_ammo", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
+
+	return true
+end
+
+---
+-- Called to drop ammo in a safe manner (e.g. preparing and space-check).
+-- @param Weapon wep The weapon that should be referenced when dropping ammo.
+-- @param boolean useClip If set to true the ammo is dropped from the clip, otherwise, from reserve ammo.
+-- @return boolean Returns if ammo is dropped
+-- @realm server
+function plymeta:SafeDropAmmo(wep, useClip)
+	if not self:CanSafeDropAmmo(wep) then
+		return false
+	end
+
+	self:AnimPerformGesture(ACT_GMOD_GESTURE_ITEM_GIVE)
```

Comment:
I think you might still want to play the animation in the weapon drop function. What did you think @TimGoll ?

Reply:
Uhm, weapon drop function? Are you talking about `WEPS.DropNotifiedAmmo`? Honestly, I don't care. But @EntranceJew did it the same way we(?) did it for weapon drops:

https://github.com/TTT-2/TTT2/blob/660d5151a1da7dd2272a154f998cb3080e1db0c2/gamemodes/terrortown/gamemode/server/sv_player_ext.lua#L1302-L1312

Reply:
Nah i was talking about the ammo drop, where the safe method should only do a safety check and the base drop function should do everything else.

Reply:
Yes, but the implementation here is fine imho, because the animation is done in the safeDrop for the weapon as well